modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/components/BreadcrumbFinder/useBreadcrumbFinder.ts

```diff
@@ -35,31 +48,66 @@ const useBreadcrumbFinder = () => {
 	const [active, setActive] = useState(false);
 	const [index, setIndex] = useState(0);
 
+	const ulRef = useRef<HTMLUListElement>(null);
+
+	const tabRefs = useMemo(
+		() =>
+			Array.from({length: items.length}).map(() =>
+				createRef<HTMLLIElement>()
+			),
+		[items.length]
+	);
+
 	const baseOptions = {enabled: active};
-	const currentEntity = defaultEntities[breadCrumb.length];
 	const tabDisabled = breadCrumb.length + 1 === defaultEntities.length;
+	const ids = breadCrumb.map(({value}) => value);
 
 	const activeItem = useMemo(
-		() => ({...items[index], entity: currentEntity}),
+		() => ({
+			...items[index],
+			entity: currentEntity,
+		}),
 		[currentEntity, index, items]
 	);
 
 	const itemsLength = items.length;
 
 	const onEnter = useCallback(() => {
-		navigate(`/project/${activeItem.value}/overview`);
-		setActive(false);
-	}, [activeItem, navigate]);
+		if (currentEntity.getPage) {
+			setBreadCrumb((prevBreadCrumb) => [...prevBreadCrumb, activeItem]);
+			setIndex(0);
+			setSearch('');
+			setActive(false);
+			navigate(currentEntity?.getPage([...ids, activeItem.value]));
+		}
+	}, [activeItem, currentEntity, ids, navigate, setBreadCrumb, setSearch]);
 
 	const onClickRow = (rowIndex: number) => {
 		if (ON_CLICK_ROW_ENABLED) {
 			const currentItem = items[rowIndex];
 
-			navigate(`/project/${currentItem.value}/overview`);
+			if (currentEntity.getPage) {
+				navigate(currentEntity?.getPage([...ids, currentItem.value]));
+				setBreadCrumb((prevBreadCrumb) => [
+					...prevBreadCrumb,
+					currentItem,
+				]);
+			}
+			setSearch('');
 			setActive(false);
 		}
 	};
 
+	const scrollIntoView = useCallback((liElement: any) => {
+		if (liElement.current) {
+			ulRef.current?.scrollTo({
+				behavior: 'smooth',
+				left: 0,
+				top: liElement.current.offsetTop - 90,
```

Comment:
Create variable for this magic number, 90

```suggestion
				top: liElement.current.offsetTop - 90,
```